packages/cli-kit/src/public/node/local-storage.ts

```diff
@@ -17,34 +20,109 @@ export class LocalStorage<T extends {[key: string]: any}> {
    *
    * @param key - The key to get.
    * @returns The value.
+   * @throws AbortError if a permission error occurs.
+   * @throws BugError if an unexpected error occurs.
    */
   get<TKey extends keyof T>(key: TKey): T[TKey] | undefined {
-    return this.config.get(key)
+    try {
+      return this.config.get(key)
+      // eslint-disable-next-line no-catch-all/no-catch-all
+    } catch (error) {
+      this.handleError(error, 'get')
+    }
   }
 
   /**
    * Set a value in the local storage.
    *
    * @param key - The key to set.
    * @param value - The value to set.
+   * @throws AbortError if a permission error occurs.
+   * @throws BugError if an unexpected error occurs.
    */
   set<TKey extends keyof T>(key: TKey, value?: T[TKey]): void {
-    this.config.set(key, value)
+    try {
+      this.config.set(key, value)
+      // eslint-disable-next-line no-catch-all/no-catch-all
+    } catch (error) {
+      this.handleError(error, 'set')
+    }
   }
 
   /**
    * Delete a value from the local storage.
    *
    * @param key - The key to delete.
+   * @throws AbortError if a permission error occurs.
+   * @throws BugError if an unexpected error occurs.
    */
   delete<TKey extends keyof T>(key: TKey): void {
-    this.config.delete(key)
+    try {
+      this.config.delete(key)
+      // eslint-disable-next-line no-catch-all/no-catch-all
+    } catch (error) {
+      this.handleError(error, 'delete')
+    }
   }
 
   /**
    * Clear the local storage (delete all values).
+   *
+   * @throws AbortError if a permission error occurs.
+   * @throws BugError if an unexpected error occurs.
    */
   clear(): void {
-    this.config.clear()
+    try {
+      this.config.clear()
+      // eslint-disable-next-line no-catch-all/no-catch-all
+    } catch (error) {
+      this.handleError(error, 'clear')
+    }
+  }
+
+  /**
+   * Handle errors from config operations.
+   * If the error is permission-related, throw an AbortError with helpful hints.
+   * Otherwise, throw a BugError.
+   *
+   * @param error - The error that occurred.
+   * @param operation - The operation that failed.
+   * @throws AbortError if the error is permission-related.
+   * @throws BugError if the error is not permission-related.
+   */
+  private handleError(error: unknown, operation: string): never {
```

Comment:
I would send the same error for all operations to simplify the code and make the Observe grouping easier. We don't care which one is failing, all of them should work.